src/lib.rs

```diff
@@ -82,38 +104,72 @@ pub fn init(interface: WlcInterface) -> Option<fn() -> ()> {
     }
 }
 
+/// Initialize wlc's callbacks and logger.
+///
+/// To initialize wlc, register your callbacks with the functions described in
+/// the `callbacks` module, and the logger using `log_set_handler` or
+/// `log_set_default_handler`. Then call `init2()`.
+///
+/// # Permissions
+/// If a compositor is initialized from the tty using suid or logind, it will
+/// drop extra permissions after a call to `init2()`. It is strongly
+/// recommended to delay code which is not registering callbacks until after
+/// this call.
+///
+/// # wlc Example
+/// ```no_run
+/// use rustwlc;
+/// use rustwlc::callback;
+/// use rustwlc::handle::WlcView;
+///
+/// // An example callback function
+/// // See the various functions in this module for more information
+/// extern "C" fn view_focus_callback(view: WlcView, focused: bool) {
+///     println!("A view came into focus!");
+/// }
+///
+/// // Set a default log callback
+/// rustwlc::log_set_default_handler();
+///
+/// // Register some callbacks
+/// callback::view_focus(view_focus_callback);
+/// // ... and additional callbacks
+///
+/// // The only thing your code should do before init2 is register callbacks
+/// // and log handlers.
+/// let run_wlc = rustwlc::init2()
+///     .expect("Unable to initialize wlc!");
+///
+/// run_wlc();
+/// ```
+pub fn init2() -> Option<fn() -> ()> {
+    unsafe {
```

Comment:
The only unsafe thing being done here in wlc_init2 right? Can' you do

`if unsafe { wlc_init2() } {....} else {....}`?